odbc/src/api/functions.rs

```diff
@@ -1207,7 +1276,41 @@ unsafe fn sql_fetch_helper(statement_handle: HStmt, function_name: &str) -> SqlR
 
         *stmt.var_data_cache.write().unwrap() = Some(HashMap::new());
 
-        if !warnings_opt.is_empty() {
+        let mut success_with_info_encountered = false;
+
+        if stmt.bound_cols.read().unwrap().is_some() {
+            let mongo_handle_for_sql_get_data_helper = MongoHandleRef::from(statement_handle);
+
+            for (col, bound_col_info) in stmt.bound_cols.read().unwrap().as_ref().unwrap().iter() {
```

Comment:
```suggestion
        if let Some(ref bound_cols) = stmt.bound_cols.read().unwrap().as_ref() {
            let mongo_handle_for_sql_get_data_helper = MongoHandleRef::from(statement_handle);

            for (col, bound_col_info) in bound_cols.iter() {
```
I didn't try this against the compiler, but is something like this possible? To reduce the duplicate `stmt.bound_cols.read().unwrap()` calls?